flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/schema/TypedFlinkTableFunction.scala

```diff
@@ -46,8 +47,18 @@ class TypedFlinkTableFunction(
       typeFactory: RelDataTypeFactory,
       arguments: Array[AnyRef],
       argTypes: Array[Class[_]]): RelDataType = {
-    // we have determined the row type before, just convert it to RelDataType
-    typeFactory.asInstanceOf[FlinkTypeFactory].createFieldTypeFromLogicalType(
-      fromDataTypeToLogicalType(externalResultType))
+    val fieldTypes = TableEnvironment.getFieldTypes(externalResultType)
```

Comment:
I would suggest to not use `TableEnvironment.getFieldTypes`, it will be deleted soon

Reply:
OK, I would use FieldInfoUtils in api-module instead.